### PR TITLE
update auction dialog and endpoint and minor changes

### PR DESCRIPTION
**Pull request by draganaaz**
- Updated add-auction-dialog to add-update-auction-dialog to use it for both, cerating and updating
- Changed addAuctionForm to addOrUpdateAuctionForm
- Created method setDates() which returns startDateTime and EndDateTime used in onCreateAuctionClick() and onUpdateAuctionClick()
- Created dateGroup form subgroup to access start and end time
- Created updateAuction endpoint
- Changed route getMyAuctions to getUserAuctions

### CODE REVIEW

File: auction9-frontend/src/app/data-table/data-table-details/data-table-details.component.html [hunk 1]
```diff
@@ -33,13 +34,13 @@ <h4>Number of bids:
...
       </button>
 		</h4>
-
   	</div>
     <p>
       {{auction?.description}}
     </p>
   </mat-card-content>
   <mat-card-actions>
+    <mat-form-field class="input-bid">
```
Comment (dangerousScript) [inline auction9-frontend/src/app/data-table/data-table-details/data-table-details.component.html:43]: I'm not sure if this input is duplicate?

Comment (draganaaz) [reply auction9-frontend/src/app/data-table/data-table-details/data-table-details.component.html:43]: Removed.

File: auction9-backend/serverless.yml [hunk 2]
```diff
@@ -66,15 +66,24 @@ functions:
           method: get
           cors: true
 
-# return all my auctions
-  getMyAuctions:
-    handler: handler.getMyAuctions
+# return all ysers auctions
```
Comment (SvetislavMarjanovic) [inline auction9-backend/serverless.yml:69]: typo: users

File: auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.html [hunk 3]
```diff
@@ -110,11 +114,14 @@ <h1 mat-dialog-title>
...
+    <mat-error *ngIf="addOrUpdateAuctionForm.controls.endTime.hasError('required')">
       You have to pick end time
     </mat-error>
   </mat-form-field>
-  <mat-error *ngIf="addAuctionForm.hasError('startTimeLessThanEndTime')">
+  <mat-error *ngIf="addOrUpdateAuctionForm.hasError('startTimeLessThanEndTime')">
+    End time must be greater than start time
+  </mat-error>
+  <mat-error *ngIf="addOrUpdateAuctionForm.hasError('minTimeGreaterThanNow')">
     End time must be greater than start time
```
Comment (SvetislavMarjanovic) [inline auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.html:125]: It looks like wrong message info?

File: auction9-backend/serverless.yml (hunk 2 above)
Comment (draganaaz) [reply auction9-backend/serverless.yml:69]: Thank you, I'll resolve it

File: auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.html (hunk 3 above)
Comment (draganaaz) [reply auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.html:125]: I forgot to remove that line, it was part of custom validator that I'll create later

File: auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+          "date_to": endDateTime,
+          "created_by": 1
+        }).subscribe(() => {
+          this.snackBar.open('Auction created successfully.', '',
+            {
+              duration: 2000,
+              panelClass: ['light-snackbar']
+            });
+        }), (err: any) => {
+          this.snackBar.open('Unable to create auction.', '',
```
Comment (mjovanovic) [inline auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.ts:126]: We should surface the reason, if any. Not in this PR

File: auction9-backend/serverless.yml [hunk 5]
```diff
@@ -66,15 +66,24 @@ functions:
...
+          path: userAuctions
           method: get
           cors: true
 
+  # updates auction
+  updateAuction:
+    handler: handler.updateAuction
+    events:
+      - http:
+          path: updateAuction
```
Comment (mjovanovic) [inline auction9-backend/serverless.yml:83]: auctions

File: auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.ts (hunk 4 above)
Comment (draganaaz) [reply auction9-frontend/src/app/add-update-auction-dialog/add-update-auction-dialog.component.ts:126]: Okay.

**Review by SvetislavMarjanovic — APPROVED** [review]

**Pull request merged**